tests/conftest.py

```diff
@@ -38,19 +49,71 @@ def keeper(accounts):
     yield accounts[5]
 
 
-@pytest.fixture
-def token():
-    token_address = "0x6b175474e89094c44da98b954eedeac495271d0f"  # this should be the address of the ERC-20 used by the strategy/vault (DAI)
-    yield Contract(token_address)
-
-
-@pytest.fixture
-def amount(accounts, token, user):
-    amount = 10_000 * 10 ** token.decimals()
+token_addresses = {
+    "WBTC": "0x2260FAC5E5542a773Aa44fBCfeDf7C193bc2C599",  # WBTC
+    "YFI": "0x0bc529c00C6401aEF6D220BE8C6Ea1667F6Ad93e",  # YFI
+    "WETH": "0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2",  # WETH
+    "LINK": "0x514910771AF9Ca656af840dff83E8264EcF986CA",  # LINK
+    "USDT": "0xdAC17F958D2ee523a2206206994597C13D831ec7",  # USDT
+    "DAI": "0x6B175474E89094C44Da98b954EedeAC495271d0F",  # DAI
+    "USDC": "0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48",  # USDC
+}
+
+# TODO: uncomment those tokens you want to test as want
+@pytest.fixture(
+    params=[
+        # 'WBTC', # WBTC
+        # "YFI",  # YFI
+        # "WETH",  # WETH
+        # 'LINK', # LINK
+        # 'USDT', # USDT
+        # 'DAI', # DAI
+        # 'USDC', # USDC
+    ],
+    scope="session",
+    autouse=True,
+)
+def token(request):
+    yield Contract(token_addresses[request.param])
+
+
+whale_addresses = {
+    "WBTC": "0x28c6c06298d514db089934071355e5743bf21d60",
```

Comment:
Found this API you can use to fetch token holders.

https://github.com/EverexIO/Ethplorer/wiki/Ethplorer-API#get-top-token-holders

Reply:
this one is useful. will look into it